### PR TITLE
Update DualCore example

### PR DESCRIPTION
**Pull request by VladimirUmek**
- update for CMSIS-Toolbox 2.6.0
- use latest packs
- add linker scripts for AC6/GCC

### CODE REVIEW

File: DualCore/HelloWorld.csolution.yml [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
   packs:
-    - pack: ARM::CMSIS
-    - pack: ARM::CMSIS-RTX
-    - pack: ARM::CMSIS-Compiler                # For retargeting stdout to UART      
-    - pack: NXP::K32L3A60_DFP@17.0.0
-    - pack: NXP::FRDM-K32L3A6_BSP@17.0.0 
+    - pack: ARM::CMSIS@^6.1.0
+    - pack: ARM::CMSIS-RTX@^5.9.0
+    - pack: ARM::CMSIS-Compiler@^2.1.0
+    - pack: NXP::K32L3A60_DFP@19.0.0
```
Comment (jkrech) [inline DualCore/HelloWorld.csolution.yml]: Should we use @^19.0.0 with caret as well. Usually NXP releases major version increments where they are "free" to be incompatible but for minor or patch updates they are required to be compatible.

File: DualCore/cm4/HelloWorld_cm4.cproject.yml [hunk 2]
```diff
@@ -24,33 +10,34 @@ project:
...
+    - component: Device:SDK Drivers:lists
+    - component: Device:SDK Drivers:lpuart
+    - component: Device:SDK Drivers:lpuart_adapter
+    - component: Device:SDK Drivers:msmc
+    - component: Device:SDK Drivers:mu
+    - component: Device:SDK Drivers:port
+
+    - component: Device:SDK Utilities:assert
+    - component: Device:SDK Utilities:debug_console
+    - component: Device:SDK Utilities:K32L3A60_linker
```
Comment (jkrech) [inline DualCore/cm4/HelloWorld_cm4.cproject.yml]: from which component are the linker scripts contributed? Is it the template?

File: DualCore/cm4/HelloWorld_cm4.cproject.yml [hunk 3]
```diff
@@ -60,12 +47,17 @@ project:
...
         - file: ../middleware/multicore/mcmgr/src/mcmgr.h
         - file: ../middleware/multicore/mcmgr/src/mcmgr_internal_core_api.h
         - file: ../middleware/multicore/mcmgr/src/mcmgr_internal_core_api_k32l3a6.c
         - file: ../middleware/multicore/mcmgr/src/mcmgr_mu_internal.c
+
+  linker:
+    - script: ./RTE/Device/K32L3A60VPJ1A_cm4/K32L3A60xxx_cm4_flash.scf
+      for-compiler: AC6
+    - script: ./RTE/Device/K32L3A60VPJ1A_cm4/K32L3A60xxx_cm4_flash.ld
+      for-compiler: GCC
```
Comment (jkrech) [inline DualCore/cm4/HelloWorld_cm4.cproject.yml]: should we have an empty last line in the files?

**Review by jkrech — CHANGES_REQUESTED** [review]
@VladimirUmek  can you please check my comments?

File: DualCore/HelloWorld.csolution.yml (hunk 1 above)
Comment (VladimirUmek) [reply DualCore/HelloWorld.csolution.yml]: Latest version uses @^ for device specific packs too.

File: DualCore/cm4/HelloWorld_cm4.cproject.yml (hunk 2 above)
Comment (VladimirUmek) [reply DualCore/cm4/HelloWorld_cm4.cproject.yml]: K32L3A60_linker provided linker scripts but I've removed this and project now uses regions header file.

File: DualCore/cm4/HelloWorld_cm4.cproject.yml (hunk 3 above)
Comment (VladimirUmek) [reply DualCore/cm4/HelloWorld_cm4.cproject.yml]: Sure.

**Review by jkrech — APPROVED** [review]
LGTM

**Pull request merged**